finetune/encoding.py

```diff
@@ -119,32 +125,49 @@ def bpe(self, token):
         self.cache[token] = word
         return word
 
-    def encode(self, texts, verbose=True):
+    def _encode(self, texts, verbose=True):
         """
         Convert a batch of raw text to a batch of byte-pair encoded token indices.
         """
         batch_token_idxs = []
-
+        batch_label_idxs = []
+        batch_character_locs = []
+        label = None
         if verbose:
             texts = tqdm(texts, ncols=80, leave=False)
 
         for text in texts:
+            if type(text) != str and len(text) == 2:
+                text, label = text
+
+            raw_text = text.lower()
             text = self.nlp(text_standardize(text))
             token_idxs = []
+            tok_pos = []
             for token in text:
+                bpe_toks = self.bpe(token.text).split()
                 token_idxs.extend([
                     self.encoder.get(t, self.UNK_IDX)
-                    for t in self.bpe(token.text).split()
+                    for t in bpe_toks
                 ])
+                token_idx = raw_text.find(token.text, tok_pos[-1] if tok_pos else 0)
+                assert len("".join(bpe_toks).replace("</w>", "")) == len(token), "{}, {}".format(bpe_toks, token)
```

Comment:
Have some questions here for in person

Reply:
In short -- make sure that we don't skip over the next token by overcounting the length of bp's.   Maybe this can be done by setting the start idx of the next call to raw_text.find() to `token_idx + len(token)` rather than `tok_pos[-1]`?